### ISSUE TITLE
OperationalContext - seperate logic from presentaion

### ISSUE THREAD

**Issue by JoshRosenstein** [contiamo/operational-ui#944]
After realizing i couldn't use some components without having them wrapped in OperationalUI and then reading over [useOperationalContext Feature](https://github.com/contiamo/operational-ui/pull/911),

Whenever you choose to refactor some of the components like (Button, Breadcrumb,Logo etc), I think it would be best to separate out the logic from UI components. I dont need to preach the reasons why, but i thought after seeing that PR, that would be a good time to do it when those get refactored.

Also you can even take it a step further and breakdown and export everything like [Uber/baseui](https://github.com/uber-web/baseui/tree/master/src/button-group) 😄

**Comment by TejasQ** [comment]
Great idea! Do you have a little bit more concrete information about which components would be touched?

**Comment by JoshRosenstein** [comment]
- [Breadcrumb](https://github.com/contiamo/operational-ui/blob/master/src/Breadcrumb/Breadcrumb.tsx#L41)
- [Logo](https://github.com/contiamo/operational-ui/blob/master/src/Logo/Logo.tsx#L68)
- [Input.Button](https://github.com/contiamo/operational-ui/blob/master/src/Input/Input.Button.tsx)
- [SimpleLink](https://github.com/contiamo/operational-ui/blob/master/src/SimpleLink/SimpleLink.tsx)
- [SidenavItem](https://github.com/contiamo/operational-ui/blob/master/src/SidenavItem/SidenavItem.tsx)
- [Button](https://github.com/contiamo/operational-ui/blob/master/src/Button/Button.tsx)
- [SidenavHeader](https://github.com/contiamo/operational-ui/blob/master/src/SidenavHeader/SidenavHeader.tsx)

- off topic but just 👀simpleLink, and wonder if there are more components like this where you should pull the container out from under and make use of the (as-prop)[https://emotion.sh/docs/styled#as-prop]

**Comment by TejasQ** [comment]
This is a great idea! It's actually a continuation (IMO) of #911. Let's do it and HOOKS ALL THE THINGS! Wanna collab?

**Comment by JoshRosenstein** [comment]
Yeah saw that and why i brought it up as it would be a perfect time to knock out 2 birds with one stone. I gotta get used to correctly linking my refs haha.

**Comment by TejasQ** [comment]
I think this is closable since we merged #911, right, @JoshRosenstein?
